components/ephemeral_storage/ephemeral_storage_service.cc

```diff
@@ -231,25 +239,22 @@ bool EphemeralStorageService::FirstPartyStorageAreaNotInUse(
 
 void EphemeralStorageService::CleanupTLDEphemeralAreaByTimer(
     const TLDEphemeralAreaKey& key,
-    bool cleanup_first_party_storage_area,
-    TLDEphemeralAreaOnDestroyCallbacks on_destroy_callbacks) {
+    bool cleanup_first_party_storage_area) {
   DVLOG(1) << __func__ << " " << key.first << " " << key.second;
   tld_ephemeral_areas_to_cleanup_.erase(key);
-  CleanupTLDEphemeralArea(key, cleanup_first_party_storage_area,
-                          std::move(on_destroy_callbacks));
+  CleanupTLDEphemeralArea(key, cleanup_first_party_storage_area);
 }
 
 void EphemeralStorageService::CleanupTLDEphemeralArea(
     const TLDEphemeralAreaKey& key,
-    bool cleanup_first_party_storage_area,
-    TLDEphemeralAreaOnDestroyCallbacks on_destroy_callbacks) {
```

Comment:
these callbacks was a hacky approach from the start. Observer implementation works better and is much cleaner.